### PR TITLE
add edit option and rofi screenshot applet

### PR DESCRIPTION
**Pull request by MrDuartePT**
Add edit option to grimblast and copy of my rofi screenshot applet script you can see both in action on this video:
https://raw.githubusercontent.com/MrDuartePT/contrib/main/grimblast/preview%20grimblast%20features.mp4

### CODE REVIEW

File: grimblast/grimblast [hunk 1]
```diff
@@ -24,6 +24,20 @@ getTargetDirectory() {
...
+#Detect if $GRIMPBLAST_EDITOR env exist
+env_editor_confirm () {
+  if [ "printenv | grep $GRIMPBLAST_EDITOR" ]; then
+    echo "GRIMPBLAST_EDITOR env is set continue...."
+  else
+    echo "GRIMPBLAST_EDITOR env is not set defaulting to gimp"
+    GRIMPBLAST_EDITOR=gimp
+  fi
+}
+
```
Comment (fufexan) [inline grimblast/grimblast:40]:
```suggestion
env_editor_confirm () {
  if [ "printenv | grep $GRIMBLAST_EDITOR" ]; then
    echo "GRIMBLAST_EDITOR is set. Continuing..."
  else
    echo "GRIMBLAST_EDITOR is not set. Defaulting to gimp"
    GRIMBLAST_EDITOR=gimp
  fi
}

```

File: grimblast/grimblast [hunk 2]
```diff
@@ -48,17 +62,19 @@ done
...
-  echo "  grimblast [--notify] [--cursor] (copy|save|copysave) [active|screen|output|area] [FILE|-]"
+  echo "  grimblast [--notify] [--cursor] (copy|save|copysave|edit) [active|screen|output|area] [FILE|-]"
   echo "  grimblast check"
   echo "  grimblast usage"
   echo ""
   echo "Commands:"
   echo "  copy: Copy the screenshot data into the clipboard."
   echo "  save: Save the screenshot to a regular file or '-' to pipe to STDOUT."
   echo "  copysave: Combine the previous 2 options."
+  echo "  edit: Open screenshot in image editor of your choice [default is gimp]. (see man page to change it)"
```
Comment (fufexan) [inline grimblast/grimblast:77]:
```suggestion
  echo "  edit: Open screenshot in the image editor of your choice (default is gimp). See man page for info."
```

File: grimblast/grimblast [hunk 3]
```diff
@@ -168,6 +184,17 @@ elif [ "$ACTION" = "save" ] ; then
   else
     notifyError "Error taking screenshot with grim"
   fi
+elif [ "$ACTION" = "edit" ] ; then
+  env_editor_confirm
+  if takeScreenshot "$FILE_EDITOR" "$GEOM" "$OUTPUT"; then
+    TITLE="Screenshot of $SUBJECT"
+    MESSAGE="Open Screenshot in Gimp (save in another location after)"
```
Comment (fufexan) [inline grimblast/grimblast:191]:
```suggestion
    MESSAGE="Open screenshot in image editor"
```

File: grimblast/grimblast.1.scd [hunk 4]
```diff
@@ -27,6 +27,15 @@ grimblast - a helper for screenshots within hyprland
...
 	Copy the screenshot data (as image/png) into the clipboard.
 
+*copysave* 
+	Combine the previous 2 options.
+
+*edit*
+	Open screenshot in image editor of your choice. The default is
+	gimp but can be edit by enviroment variable $GRIMPBLAST_EDITOR
+	(this can be change by adding export GRIMPBLAST_EDITOR=[editor command] 
+	on your .bashrc or equivalent) [Exemple:export GRIMPBLAST_EDITOR=gimp]
```
Comment (fufexan) [inline grimblast/grimblast.1.scd:37]:
```suggestion
	Open screenshot in the image editor of your choice. The default is
	gimp, but you can set a different one with the enviroment variable
	$GRIMBLAST_EDITOR. Example: `export GRIMBLAST_EDITOR=gimp`.
```

File: grimblast/screenshot.sh [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+option_6=" Capture in 10s"
+
+list_col='1'
+list_row='6'
+win_width='300px'
+
+copy=' Copy'
+save=' Save'
+copy_save='Copy & Save'
+gimp='Edit Screenshot'
```
Comment (fufexan) [inline grimblast/screenshot.sh:35]:
```suggestion
edit='Edit Screenshot'
```

File: grimblast/screenshot.sh [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+# CMD
+copy_save_gimp_cmd () {
+	rofi -theme-str 'window {location: center; anchor: center; fullscreen: false; width: 400px;}' \
+		-theme-str 'mainbox {orientation: vertical; children: [ "message", "listview" ];}' \
+		-theme-str 'listview {columns: 2; lines: 2;}' \
+		-theme-str 'element-text {horizontal-align: 0.5;}' \
+		-theme-str 'textbox {horizontal-align: 0.5;}' \
+		-dmenu \
+		-p 'Choose Option' \
+		-mesg 'You whant to copy or/and save the screenshot or open in Gimp'
```
Comment (fufexan) [inline grimblast/screenshot.sh:65]:
```suggestion
		-mesg 'Copy/save the screenshot or open in image editor'
```

File: grimblast/screenshot.sh [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+# Ask for confirmation
+copy_save_gimp_exit() {
```
Comment (fufexan) [inline grimblast/screenshot.sh:69]:
```suggestion
copy_save_editor_exit() {
```

File: grimblast/screenshot.sh [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+	echo -e "$copy\n$save\n$copy_save\n$gimp" | copy_save_gimp_cmd
```
Comment (fufexan) [inline grimblast/screenshot.sh:70]:
```suggestion
	echo -e "$copy\n$save\n$copy_save\n$edit" | copy_save_edit_cmd
```

File: grimblast/screenshot.sh [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+		option_chose=save
+    	${1}
+	elif [[ "$selected" == "$copy_save" ]]; then
+		option_chose=copysave
+        ${1}
+    elif [[ "$selected" == "$gimp" ]]; then
+        option_chose=edit
+        ${1}
+    fi	
+}
```
Comment (fufexan) [inline grimblast/screenshot.sh:89]:
```suggestion
copy_save_edit_run () {	
	selected="$(copy_save_edit_exit)"
	if [[ "$selected" == "$copy" ]]; then
		option_chosen=copy
    	${1}
	elif [[ "$selected" == "$save" ]]; then
		option_chosen=save
    	${1}
	elif [[ "$selected" == "$copy_save" ]]; then
		option_chosen=copysave
        ${1}
    elif [[ "$selected" == "$edit" ]]; then
        option_chosen=edit
        ${1}
    fi	
}
```

File: grimblast/screenshot.sh [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	elif [[ "$1" == '--opt2' ]]; then
+		copy_save_gimp_run "shotdesktop_active"
+	elif [[ "$1" == '--opt3' ]]; then
+		copy_save_gimp_run "shotarea"
+	elif [[ "$1" == '--opt4' ]]; then
+		copy_save_gimp_run "shotwin"
+	elif [[ "$1" == '--opt5' ]]; then
+		copy_save_gimp_run "shot5"
+	elif [[ "$1" == '--opt6' ]]; then
+		copy_save_gimp_run "shot10"
```
Comment (fufexan) [inline grimblast/screenshot.sh:143]: `copy_save_gimp_run` -> `copy_save_edit_run`

**Review by fufexan — CHANGES_REQUESTED** [review]
Apart from the changes requested, please also add `grimblast/screenshot.sh` to `.github/workflows/check.yml` and fix any issues found by shellcheck.

**Comment by MrDuartePT** [comment]
Thanks for the notice
I will run the workflow and update the pull request

**Comment by MrDuartePT** [comment]
Spellcheck run fine i just have to comment out the env_editor_confirm function because spellcheck fail to run with this code.
But besides that the check runs fine.
image

**Comment by fufexan** [comment]
> Spellcheck run fine i just have to comment out the env_editor_confirm function because spellcheck fail to the frist time with this code.

You have to change the condition to `if [ $(printenv GRIMBLAST_EDITOR; echo $?) -eq 0 ]; then`

**Comment by MrDuartePT** [comment]
Now is all check :)
Thanks for your help and sorry for spelling mistake on the script.
I do some improvements to the screenshot script if you want i can do another video showing the changes.

File: README.md [hunk 11]
```diff
@@ -21,3 +21,9 @@ See `shellevents_default.sh` for the supported function names. Example event fil
 Contributions are welcome! Please open a PR with your utility and describe how
 it helps the Hypr ecosystem. Also provide a screen recording if possible,
 demonstrating its usage.
+
+# Changelog
+
+Add 'edit' to Grimpblast option read the [man page](grimblast/grimblast.1.scd) for more information
+
+Copy of applet screenshot script for rofi
```
Comment (fufexan) [inline README.md:29]:
```suggestion
Added the `edit` option to Grimblast. read the [man page](grimblast/grimblast.1.scd) for more information.
A rofi script is also included.
```

**Comment by fufexan** [comment]
Thanks for fixing the above issues. Can you squash the commits into one (everything) or maybe two (grimblast part in one commit, rofi script in another)?

**Comment by MrDuartePT** [comment]
Done

**Comment by MrDuartePT** [comment]
Done add your README.md suggestion and squash everything in one commit.

**Review by fufexan — APPROVED** [review]
LGTM

**Pull request merged**